dkg/src/epoch_manager.rs

```diff
@@ -162,10 +167,25 @@ impl<P: OnChainConfigProvider> EpochManager<P> {
             .get(&self.my_addr)
             .copied();
 
-        let onchain_randomness_config = payload
-            .get::<RandomnessConfigMoveStruct>()
-            .and_then(OnChainRandomnessConfig::try_from)
-            .unwrap_or_else(|_| OnChainRandomnessConfig::default_if_missing());
+        let onchain_randomness_config_seq_num = payload
+            .get::<RandomnessConfigSeqNum>()
+            .unwrap_or_else(|_| RandomnessConfigSeqNum::default_if_missing());
+
+        info!("local_seqnum={}", self.randomness_override_seq_num);
+        info!(
+            "onchain_seqnum={}",
+            onchain_randomness_config_seq_num.seq_num
+        );
+        let onchain_randomness_config =
+            if onchain_randomness_config_seq_num.seq_num < self.randomness_override_seq_num {
+                info!("Randomness emergency stop.");
+                OnChainRandomnessConfig::default_disabled()
+            } else {
+                payload
+                    .get::<RandomnessConfigMoveStruct>()
+                    .and_then(OnChainRandomnessConfig::try_from)
+                    .unwrap_or_else(|_| OnChainRandomnessConfig::default_if_missing())
+            };
```

Comment:
feels like this code block can be reused, adding a function in the OnChainRandomnessConfig to take payload and local seqnum as params?

Reply:
fixed